routers/web/repo/view.go

```diff
@@ -58,7 +58,7 @@ const (
 //	entries == ctx.Repo.Commit.SubTree(ctx.Repo.TreePath).ListEntries()
 //
 // FIXME: There has to be a more efficient way of doing this
-func findReadmeFileInEntries(ctx *context.Context, entries []*git.TreeEntry) (string, *git.TreeEntry, error) {
+func findReadmeFileInEntries(ctx *context.Context, entries []*git.TreeEntry, recurse bool) (string, *git.TreeEntry, error) {
```

Comment:
This is a really unsatisfying way to solve this, to me, but it works.

Pre-https://github.com/go-gitea/gitea/pull/22177, this was avoided because one of the variant versions of `findReadmeFile()` called into the other, which didn't, so there was never a full recursion. Perhaps that was the hidden wisdom in that code.

I also had this working before by recursing on `treePath`:

https://github.com/go-gitea/gitea/blob/13ab8a1a7faaf3d702a0c8bddba23442b916cf09/routers/web/repo/view.go#L60-L61

but in https://github.com/go-gitea/gitea/pull/22177#discussion_r1053953034 that was dropped because Lunny didn't want to call ListEntries() repeatedly. By working in terms of `git.TreeEntry` instead of a path string, there is no way to tell if we're at the root or not anymore, because Git Tree Entries don't know what their parent is (in fact they don't have a parent, but multiple parents can have them as children, which is how git deduplicates content shared between commits/folders)

So fixing it with stupid bool flag is the best option I've found.

Reply:
I guess the `recurse` doesn't mean what the word's meaning.

I think it means something like:

* `atRoot`:  if at root, then try to find `/docs/`, etc
* `tryWellKnownDirs`:  whether the call should try to find well-known dirs like `/docs/` at current path.

Reply:
That's good enough for me. I've changed the name.